media/PartEditor/index.js

```diff
@@ -124,7 +135,87 @@ editor.Editor = class {
       }
     };
 
-    // TODO request partition information
+    this.requestPartition();
+  }
+
+  /**
+   * @brief apply document partition to operators listbox
+   */
+  handleResultPartition(message) {
+    this.partition = message.part;
+
+    this.updateOperatorsBackend();
+    this.refershOpListbox();
+  }
+
+  /**
+   * @brief apply document partition to this.operators code
+   */
+  updateOperatorsBackend() {
+    // must clear first as this.partition.OPNAME may not have all nodes
+    this.clearOperatorsCode();
+
+    for (let name in this.partition.OPNAME) {
+      if (this.partition.OPNAME.hasOwnProperty(name)) {
```

Comment:
Does this if condition ever become 'false'? Because `name` is an element of for loop that iterates `this.partition.OPNAME`.

Reply:
You are right! I think this is some trace while implementation.